src/Pages/UserPages/Landing/Landing.tsx

```diff
@@ -1,19 +1,292 @@
+import { Box, Button, Popover, TextField } from '@mui/material';
+import Typography from '@mui/material/Typography';
+import { DateRangeCalendar } from '@mui/x-date-pickers-pro/DateRangeCalendar';
+import { AdapterDayjs } from '@mui/x-date-pickers/AdapterDayjs';
+import { LocalizationProvider } from '@mui/x-date-pickers/LocalizationProvider';
+import { DemoContainer } from '@mui/x-date-pickers/internals/demo';
+import dayjs, { Dayjs, Range } from 'dayjs';
+import { useCallback, useEffect, useState } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import './Landing.module.scss';
+import { AddFavoriteItem } from '@/Redux/Features/Portal/Favorites/AddToFavoriteSlice';
+import { toast } from 'react-toastify';
+import { RemoveFavoriteItem } from '@/Redux/Features/Portal/Favorites/RemoveFavoriteItemSlice';
+import { getFavoriteItems } from '@/Redux/Features/Portal/Favorites/GetAllFavoritesSlice';
+import { useNavigate } from 'react-router-dom';
 const Landing = () => {
 
   const dispatch = useDispatch();
+  const { count } = useSelector((state) => state.AddToFavorite)
+  const { data } = useSelector((state) => state.RemoveFavoriteItemSlice)
+
+  const [personsCount, setPersonsCount] = useState(1);
+const navigate =  useNavigate()
+
+  const [anchorEl, setAnchorEl] = useState(null);
+  const [selectedDateRange, setSelectedDateRange] = useState<Range<Dayjs>>([
+    dayjs('2024-01-25'),
+    dayjs('2024-01-30'),
+  ]);
+console.log(selectedDateRange);
+console.log(personsCount);
+
 
   useEffect(() => {
+    getAdsData()
     dispatch(fetchDataIslogged());
+    getFavoriteData()
+  }, [dispatch,count,data]);
+
+
+  const handleCalendarChange = (newDateRange: Range<Dayjs>) => {
+    setSelectedDateRange(newDateRange);
+  };
+
+  const handleButtonClick = (event) => {
+    setAnchorEl(event.currentTarget);
+  };
+
+  const handlePopoverClose = () => {
+    setAnchorEl(null);
+  };
+
+  const handleDateSelected = () => {
+    handlePopoverClose();
+  };
+
+  const handleIncrease = () => {
+    setPersonsCount(personsCount + 1);
+  };
+
+  const handleDecrease = () => {
+    if (personsCount > 1) {
+      setPersonsCount(personsCount - 1);
+    }
+  };
+
+
+const startDate =selectedDateRange[0]?.format('YYYY-MM-DD')
+const endDate =selectedDateRange[1]?.format('YYYY-MM-DD')
+
+  console.log(`${selectedDateRange[0]?.format('YYYY-MM-DD')} - ${selectedDateRange[1]?.format('YYYY-MM-DD')}`);
+  const [adsData, setAdsData] = useState();
+  //! ************************ Rooms Ads *************************
+  const getAdsData = useCallback(async () => {
+
+    try {
+      // @ts-ignore
+      const element = await dispatch(AllAdsData());
+      // @ts-ignore
+      setAdsData(element?.payload?.data?.data?.rooms);
+    } finally {
+
+    }
   }, [dispatch]);
 
+
+
+
+  //! ************************ Add To Favorite  *************************
+  const addItemToFavorite = useCallback(async (roomId: any) => {
```

Comment:
There is no benefit memoizing this function, as it is neither complicated nor we pass it to a child component.